lib/promise/group.rb

```diff
@@ -0,0 +1,39 @@
+class Promise
+  class Group
+    attr_reader :promise
+
+    def initialize(inputs)
+      @remaining = 0
+      @promise = Promise.new
+      @inputs = inputs
+      @on_fulfill = method(:on_fulfill)
+      @on_reject = promise.method(:reject)
```

Comment:
Yup, I just did this for memoization since each call to method returns a new object.